packages/jitm/src/class-jitm.php

```diff
@@ -25,28 +24,26 @@ class JITM {
 
 	const PACKAGE_VERSION = '1.0'; // TODO: Keep in sync with version specified in composer.json.
 
-	/**
-	 * Tracking object.
-	 *
-	 * @var Automattic\Jetpack\Tracking
-	 *
-	 * @access private
-	 */
-	private $tracking;
-
 	/**
 	 * The configuration method that is called from the jetpack-config package.
 	 */
 	public static function configure() {
-		$jitm = new self();
+		$jitm = self::get_instance();
 		$jitm->register();
 	}
 
 	/**
-	 * JITM constructor.
+	 * Pre/Post Connection JITM factory metod
+	 *
+	 * @return Post_Connection_JITM|Pre_Connection_JITM JITM instance.
 	 */
-	public function __construct() {
-		$this->tracking = new Tracking();
+	public static function get_instance() {
+		if ( \Jetpack::is_active() ) {
+			$jitm = new Post_Connection_JITM();
+		} else {
+			$jitm = new Pre_Connection_JITM();
+		}
```

Comment:
I would recommend bailing and not displaying anything when `Status() )->is_development_mode()`, as folks cannot connect to WordPress.com and won't really be able to act on the pre-connection messages.

Reply:
Implemented here https://github.com/Automattic/jetpack/pull/15718/files#diff-c1bc1500616d8175c2303c8609a02c96R69